### ISSUE TITLE
Manual timer incorrectly triggered on Luup reload

### ISSUE THREAD

**Issue by brianewman** [toggledbits/DelayLight#14]
My DelayLight is configured for "Home" mode only, but it always goes into manual mode (and turns on my lamp) when my daily "restart Luup" scene runs in the middle of the night (in "Night" mode).

I think the fix is to check house mode on line 1176 in L_DelayLight.lua, like so...

```
    -- Pick up where we left off before restart...
    if status ~= STATE_IDLE then
        L("Timer %2 (%3) Continuing %1 timing across restart...", status, tdev, luup.devices[tdev].description)
        setMessage("Recovering from reload", tdev)
    elseif isEnabled( tdev ) and isActiveHouseMode( tdev ) then
       -- We think we're idle/off, but check to see if we missed events during reboot/reload
```

**Comment by toggledbits** [comment]
The `trigger()` function that is called should be checking house mode. Can you post your config? Follow the "Plugin Status" link at the bottom of the settings page.

**Comment by brianewman** [comment]
trigger() function appears to check house mode only if state == STATE_AUTO (line 1003).  In the else part, at line 1020, house mode is not checked.  

        if state == STATE_AUTO then
            if not isActiveHouseMode( tdev ) then
                D("trigger() not in an active house mode, not triggering")
                -- Not an active house mode; do nothing.
                return
            end
            offDelay = getVarNumeric( "AutoDelay", 60, tdev, TIMERSID )
            if offDelay == 0 then return end -- 0 delay means no auto-on function
            onDelay = getVarNumeric( "OnDelay", 0, tdev, TIMERSID )
            if onDelay == 0 then
                luup.variable_set( TIMERSID, "OnTime", 0, tdev, TIMERSID )
            else
                luup.variable_set( TIMERSID, "OnTime", os.time() + onDelay, tdev )
                timing = 2
                D("trigger() configuring on delay %1 seconds", onDelay)
            end
        else
            -- Trigger manual
            offDelay = getVarNumeric( "ManualDelay", 3600, tdev, TIMERSID )
            if offDelay == 0 then return end -- 0 delay means no manual delay function
            luup.variable_set( TIMERSID, "OnTime", 0, tdev )
        end

So maybe the fix is to move the "if not isActiveHouseMode" part outside of (before) the "if state == STATE_AUTO" part.  


Here's my config:
{ "configversion": 109, "type": "urn:schemas-toggledbits-com:device:DelayLight:1", "author": "Patrick H. Rigney (rigpapa)", "system": { "isOpenLuup": false, "isALTUI": true, "version": "1.7.4000", "units": "F" }, "name": "DelayLight", "devices": [ { "type": "urn:schemas-toggledbits-com:device:DelayLightTimer:1", "description": "DelayLight Timer 1", "id": "d313t1", "devNum": 314, "device_json": "D_DelayLightTimer_UI7.json", "timerState": { "off": { "21": { "valueOn": [ { "value": 0, "comparison": ">" } ], "watched": true, "variable": "LoadLevelStatus", "list": "off", "device": 21, "service": "urn:upnp-org:serviceId:Dimming1", "invert": false, "devicename": "Bedroom Light" }, "85": { "valueOn": [ { "value": 0, "comparison": ">" } ], "watched": true, "variable": "LoadLevelStatus", "list": "off", "device": 85, "service": "urn:upnp-org:serviceId:Dimming1", "invert": false, "devicename": "Bedroom Lamp" }, "22": { "valueOn": [ { "value": 0, "comparison": ">" } ], "watched": true, "variable": "LoadLevelStatus", "list": "off", "device": 22, "service": "urn:upnp-org:serviceId:Dimming1", "invert": false, "devicename": "Bedroom Fan" } }, "eventList": [ { "state": "auto", "time": "20181216T211859", "when": 1545016739, "event": "trigger", "dev": 314 }, { "state": "auto", "time": "20181216T211900", "when": 1545016740, "event": "trigger", "dev": 314 }, { "state": "auto", "time": "20181216T211901", "when": 1545016741, "event": "trigger", "dev": 314 }, { "state": "auto", "time": "20181216T211902", "when": 1545016742, "event": "trigger", "dev": 314 }, { "state": "auto", "time": "20181216T211903", "when": 1545016743, "event": "trigger", "dev": 314 }, { "state": "auto", "time": "20181216T211904", "when": 1545016744, "event": "trigger", "dev": 314 }, { "state": "auto", "time": "20181216T211905", "when": 1545016745, "event": "trigger", "dev": 314 }, { "state": "auto", "time": "20181216T211906", "when": 1545016746, "event": "trigger", "dev": 314 }, { "state": "auto", "time": "20181216T211907", "when": 1545016747, "event": "trigger", "dev": 314 }, { "state": "auto", "time": "20181216T211908", "when": 1545016748, "event": "trigger", "dev": 314 }, { "state": "auto", "time": "20181216T211909", "when": 1545016749, "event": "trigger", "dev": 314 }, { "state": "auto", "time": "20181216T211910", "when": 1545016750, "event": "trigger", "dev": 314 }, { "state": "auto", "time": "20181216T211911", "when": 1545016751, "event": "trigger", "dev": 314 }, { "state": "auto", "time": "20181216T211912", "when": 1545016752, "event": "trigger", "dev": 314 }, { "state": "auto", "time": "20181216T211913", "when": 1545016753, "event": "trigger", "dev": 314 }, { "state": "auto", "time": "20181216T211914", "when": 1545016754, "event": "trigger", "dev": 314 }, { "state": "auto", "time": "20181216T211915", "when": 1545016755, "event": "trigger", "dev": 314 }, { "state": "auto", "time": "20181216T211916", "when": 1545016756, "event": "trigger", "dev": 314 }, { "state": "auto", "time": "20181216T211917", "when": 1545016757, "event": "trigger", "dev": 314 }, { "state": "auto", "time": "20181216T211918", "when": 1545016758, "event": "trigger", "dev": 314 }, { "state": "auto", "time": "20181216T211919", "when": 1545016759, "event": "trigger", "dev": 314 }, { "state": "auto", "time": "20181216T211920", "when": 1545016760, "event": "trigger", "dev": 314 }, { "state": "auto", "time": "20181216T211921", "when": 1545016761, "event": "trigger", "dev": 314 }, { "state": "auto", "time": "20181216T211922", "when": 1545016762, "event": "trigger", "dev": 314 }, { "state": "auto", "time": "20181216T211923", "when": 1545016763, "event": "trigger", "dev": 314 } ], "pollList": [  ], "inhibit": { "327": { "valueOn": "1", "watched": true, "variable": "Tripped", "list": "inhibit", "device": 327, "service": "urn:micasaverde-com:serviceId:SecuritySensor1", "invert": true, "devicename": "Bedroom Darkness" } }, "on": { "85": { "valueOn": [ { "value": 0, "comparison": ">" } ], "watched": true, "variable": "LoadLevelStatus", "list": "on", "device": 85, "service": "urn:upnp-org:serviceId:Dimming1", "invert": false, "devicename": "Bedroom Lamp" } }, "trigger": { "323": { "valueOn": "1", "watched": true, "variable": "Tripped", "list": "trigger", "device": 323, "service": "urn:micasaverde-com:serviceId:SecuritySensor1", "invert": false, "devicename": "Bedroom Motion" } } }, "udn": "uuid:4d494342-5342-5645-013a-000002aed803", "room": 4, "impl_file": "", "model": "", "device_file": "D_DelayLightTimer.xml", "manufacturer": "", "parent": 313, "states": [ { "id": 28730, "service": "urn:toggledbits-com:serviceId:DelayLightTimer", "value": "1", "variable": "Enabled" }, { "id": 28731, "service": "urn:toggledbits-com:serviceId:DelayLightTimer", "value": "auto", "variable": "Status" }, { "id": 28732, "service": "urn:toggledbits-com:serviceId:DelayLightTimer", "value": "1", "variable": "Timing" }, { "id": 28733, "service": "urn:toggledbits-com:serviceId:DelayLightTimer", "value": "Waiting for Bedroom Motion", "variable": "Message" }, { "id": 28734, "service": "urn:toggledbits-com:serviceId:DelayLightTimer", "value": "323", "variable": "Triggers" }, { "id": 28735, "service": "urn:toggledbits-com:serviceId:DelayLightTimer", "value": "85", "variable": "OnList" }, { "id": 28736, "service": "urn:toggledbits-com:serviceId:DelayLightTimer", "value": "21,85,22", "variable": "OffList" }, { "id": 28737, "service": "urn:toggledbits-com:serviceId:DelayLightTimer", "value": "900", "variable": "AutoDelay" }, { "id": 28738, "service": "urn:toggledbits-com:serviceId:DelayLightTimer", "value": "1800", "variable": "ManualDelay" }, { "id": 28739, "service": "urn:toggledbits-com:serviceId:DelayLightTimer", "value": "0", "variable": "OnDelay" }, { "id": 28740, "service": "urn:toggledbits-com:serviceId:DelayLightTimer", "value": "2", "variable": "HoldOn" }, { "id": 28741, "service": "urn:toggledbits-com:serviceId:DelayLightTimer", "value": "1545017663", "variable": "OffTime" }, { "id": 28742, "service": "urn:toggledbits-com:serviceId:DelayLightTimer", "value": "0", "variable": "OnTime" }, { "id": 28743, "service": "urn:toggledbits-com:serviceId:DelayLightTimer", "value": "0", "variable": "ForcePoll" }, { "id": 28744, "service": "urn:toggledbits-com:serviceId:DelayLightTimer", "value": "323,1545016657,1,urn:micasaverde-com:serviceId:SecuritySensor1,Tripped", "variable": "LastTrigger" }, { "id": 28745, "service": "urn:toggledbits-com:serviceId:DelayLightTimer", "value": "1", "variable": "HouseModes" }, { "id": 28746, "service": "urn:toggledbits-com:serviceId:DelayLightTimer", "value": "-327", "variable": "InhibitDevices" }, { "id": 28747, "service": "urn:toggledbits-com:serviceId:DelayLightTimer", "value": "", "variable": "ActivePeriods" }, { "id": 28748, "service": "urn:toggledbits-com:serviceId:DelayLightTimer", "value": "1", "variable": "ManualOnScene" }, { "id": 28749, "service": "urn:toggledbits-com:serviceId:DelayLightTimer", "value": "1", "variable": "ResettableOnDelay" }, { "id": 28750, "service": "urn:toggledbits-com:serviceId:DelayLightTimer", "value": "109", "variable": "Version" }, { "id": 28751, "service": "urn:toggledbits-com:serviceId:DelayLightTimer", "value": "0", "variable": "LastPoll" }, { "id": 28752, "service": "urn:toggledbits-com:serviceId:DelayLightTimer", "value": "", "variable": "SceneData" }, { "id": 28753, "service": "urn:micasaverde-com:serviceId:HaDevice1", "value": "0", "variable": "Configured" } ] }, { "type": "urn:schemas-toggledbits-com:device:DelayLight:1", "description": "DelayLight Controlle", "id": "", "devNum": 313, "device_json": "D_DelayLight_UI7.json", "udn": "uuid:4d494342-5342-5645-0139-000002aed803", "room": 1, "impl_file": "I_DelayLight.xml", "model": "", "device_file": "D_DelayLight.xml", "manufacturer": "", "parent": 0, "states": [ { "id": 7490, "service": "urn:toggledbits-com:serviceId:DelayLight", "value": "Started 1/1 at 12/16/18 06:20:16", "variable": "Message" }, { "id": 7491, "service": "urn:toggledbits-com:serviceId:DelayLight", "value": "true", "variable": "UI7Check" }, { "id": 7492, "service": "urn:toggledbits-com:serviceId:DelayLight", "value": "0", "variable": "NumChildren" }, { "id": 7493, "service": "urn:toggledbits-com:serviceId:DelayLight", "value": "0", "variable": "NumRunning" }, { "id": 7494, "service": "urn:toggledbits-com:serviceId:DelayLight", "value": "0", "variable": "DebugMode" }, { "id": 7495, "service": "urn:toggledbits-com:serviceId:DelayLight", "value": "109", "variable": "Version" }, { "id": 7496, "service": "urn:micasaverde-com:serviceId:HaDevice1", "value": "0", "variable": "Configured" } ] } ], "timestamp": 1545016763, "url": "https://www.toggledbits.com/delaylight", "responder": 313, "version": "1.6" }

**Comment by toggledbits** [comment]
I also just noticed in your config that there may be something odd going on. The event list shows that the timer is being triggered to auto every second. This would imply that your motion sensor is sending continuous updates, which is unusual. Would you be up for enabling debug on your end and sending me a log file?

**Comment by toggledbits** [comment]
By the way, the house mode restriction only applies to auto triggering. Manual triggering/timing needs to proceed regardless of house mode so that lights are not left on indefinitely when turned on manually. Your suggested fixes would break this feature.

**Comment by brianewman** [comment]
OK, that makes sense.  It worked for me because I don't care about manual
mode at night, at least for my current use case.

I'll apply the new version and look at getting debugs a little later, maybe
tonight.

Thanks for being so responsive!

**Comment by toggledbits** [comment]
No problem. FYI, I've done pretty much nothing all day but try and duplicate your conditions. I have a test Vera set up with a sensor, inhibitor, and some dummy loads like yours. I can't make it happen. I've also done a walk-through of the code by hand, and everything seems like it should be OK. But I am on this like fur on a weasel, and I want to understand it and get it working as expected/desired.

I think the key is that we have to capture what happens when the light comes on at the wrong time. I'm hoping you can reproduce that easily, without having to wake yourself at 2am and fuss around with lights and Vera.

To turn debug on, go into the DelayLight master (plugin) device, into _Advanced > Variables_, and set the `DebugMode` variable to 1. The restart Luup, and let the fun begin. It logs a LOT of debug, but that should be helpful.

Triggering into manual means one of your loads is being reported "on" by Vera at the time of reload. The question then may be, are all your switches "instant status?" Because any switch that isn't instant status relies on Vera polling for update when operated manually, and that may factor in to the problem (Vera's polling algorithm is so bad that devices with even the minimum 60 second polling interval can be skipped for _hours_). Anyway, whatever the cause, there's a setting (timer state variable) called `ManualOnScene` which controls whether the "on" list devices (just #85 Bedroom Lamp in your case) are turned on when manual triggering is initiated. Setting `ManualOnScene=0` will cause DelayLight to _not_ turn on the "on" list devices when triggered to manual timing, which would keep the light from coming on at the wrong time. So that would probably fix the symptom you are observing, but I'd really like to get to root cause and see what else could be done.

**Comment by brianewman** [comment]
ManualOnScene=0 did indeed mask the symptom.

My Bedroom Lamp z-wave outlet does NOT have instant status, but during
testing I'm only making changes to it through Vera so its status is always
accurate.  In "real life" it's also turned off via a scene whenever Night
mode starts, so Vera should always know it's off when the automatic Luup
reload happens.  If instant status could still be a problem, it might be
one I have to live with.

Thanks,
Brian

**Comment by toggledbits** [comment]
OK. With debug on, it should be fairly clear after the reload what the setup is trying to do. It may not be the Bedroom Lamp, as ANY of the "on" or "off" list devices being thought to be "on" will trigger manual timing.

The other thing that's a bit suspect in your config is the hold-over setting. Generally speaking, mode 2 (do not start off delay until trigger device resets) is not necessary, particularly with most motion sensors. It's more intended for keeping lights on while a door remains open. The problem with using it is if that a fast reset of the sensor could prevent any triggering at all--that mode requires that the sensor still be triggered at the end of the "on" delay to have the timer turn on loads and begin the "off" delay. Unless you have a compelling reason for using mode 2, you should be in mode 0 (the default).

**Comment by brianewman** [comment]
How do I collect the debugs?  Is it in the LuaUPnP log file?

I selected mode 2 in an attempt to prevent the lights from going off while
we're still moving around the room but only triggering the motion sensor
occasionally.  I had it running while watching TV and the lamp kept going
on and off.  Maybe I just needed to increase the delay.

**Comment by brianewman** [comment]
I think I see the problem.  It's watching Bedroom Lamp and Bedroom Fan as
if it they were dimmers, rather than switches.

Line 1600: 50      12/17/18 20:44:11.492   luup_log:313:
DelayLight(debug)::: watchTriggers(): watching 85 ("Bedroom Lamp") as
dimmer <0x7684f520>
Line 1601: 50      12/17/18 20:44:11.492   luup_log:313:
DelayLight(debug):::
watchVariable("urn:upnp-org:serviceId:Dimming1","LoadLevelStatus",85,314)
<0x7684f520>
Line 1602: 50      12/17/18 20:44:11.493   luup_log:313:
DelayLight(debug)::: watchVariable() already watch in place for 314 on
"85:urn:upnp-org:serviceId:Dimming1/LoadLevelStatus" <0x7684f520>
Line 1603: 50      12/17/18 20:44:11.494   luup_log:313:
DelayLight(debug)::: watchTriggers(): watching 22 ("Bedroom Fan") as dimmer
<0x7684f520>
Line 1604: 50      12/17/18 20:44:11.505   luup_log:313:
DelayLight(debug):::
watchVariable("urn:upnp-org:serviceId:Dimming1","LoadLevelStatus",22,314)
<0x7684f520>

LoadLevelStatus is always 100 on these devices so it thinks they're on.  I
think the problem is that they both respond to dimmer commands even though
they're not dimmers and they trick your isDimmerType function.  These are
not the same device, either.  Bedroom Lamp is an outlet while Bedroom Fan
is an in-wall switch.

**Comment by toggledbits** [comment]
Most motion sensors have a "reset period" that kicks in when they detect motion. They won't detect motion again during that period. Since motion sensors tend to be security devices, and the standard in the alarm industry has been 4 minutes (to prevent nuisance alerts to the central monitoring station), a lot of the sensors I run into have reset periods around that long by default (Aeotec Multisensors, for example). That works well for breach detection, but less well for occupancy: it can cause an ironic series of events where you move around during the reset period (when the sensor is not sending any messages) and stop moving right as the reset period expires, so it misses one or even several motion events that DelayLight would have used to extend the timing.

The best approach, if your sensor supports it, is to reduce this reset period to something shorter, like 60 seconds. But in a bedroom, where you may be quite still for long periods of time (watching TV, reading, etc.) you may need to play around to see what works best for you. Usually, though, a reset period on the sensor itself (part of its device configuration for many device--some via "Device Configuration", some via jumpers on the board inside the sensor) to something low, like 60 seconds or perhaps even less (I use 15 seconds in my bathrooms, hallways and stairwells). The "auto" delay time should then be set high enough to accommodate any typical "still" time during normal occupancy. When both are set properly, there is a short trigger that starts auto timing, followed by a series of short triggers each time motion is detected after, and each of these subsequent triggers restarts the timer delay.

The only caveat to this is that if you set your sensor's reset period too short in a frequently-occupied room (like a family room or kitchen), the high rate of sensor messages will eat up the battery more quickly. So for battery-operated sensors, you'll need to balance timing with battery life a bit.

**Comment by toggledbits** [comment]
OK. So I did just spot something that could be the issue. If you would, please download the linked version (stable branch) of L_DelayLight.lua, set your ManualOnScene back to 1, upload the new Lua file (with "Reload luup after upload" on), and then test again.

My suspicion that this code will test: Luup is inconsistent in its handling of LoadLevelStatus on dimmers. On most dimmers I've found, the value is 0 when the light is off, but on a rare few, the value is left at the last known dimming level (which would make it appear "on" to DelayLight's checks). This version will check only SwitchPower1/Status to determine if a switch or dimmer is on. If this works,

New version: https://raw.githubusercontent.com/toggledbits/DelayLight/stable/L_DelayLight.lua

**Comment by brianewman** [comment]
Thanks for that information.  My sensor appears as tripped in Vera for 3
minutes before it goes back to untripped state.  (adjustable from 1 to 255
minutes)  It's this one right here:
https://downloads.monoprice.com/files/manuals/15902_Manual_160722.pdf

I'll just have to play with it to see what works.  For now, it works fine
with ManualOnScene set to 0.  I would like to see that as a checkbox on the
settings page as well.  It seems useful.

FYI, here is the outlet I'm using for Bedroom Lamp:
https://www.amazon.com/gp/product/B00JFK1YRE
I don't know if always returning the LoadLevelStatus as 100 is normal for a
z-wave outlet or not, but my Linear non-dimming in-wall switches appear to
do the same thing.

**Comment by toggledbits** [comment]
We were writing at the same time, but what you wrote leads me to believe we are fast upon the real answer.

**Comment by brianewman** [comment]
Thanks.  I'll have to test this tomorrow as my wife is going to sleep now.
:-)

**Comment by toggledbits** [comment]
Yup, I'm turning in as well. Somehow, I missed your previous message that also comes to this conclusion, but nonetheless, here we are. I think a real fix is going to be straight-foward.

**Comment by brianewman** [comment]
That did the trick.  Thanks!

50      12/18/18 10:14:49.335   luup_log:313: DelayLight(debug)::: start()
checking devices in idle startup <0x778ab520>
50      12/18/18 10:14:49.336   luup_log:313: DelayLight(debug):::
isMapDeviceOn({ 323={ valueOn="1", watched=true, variable="Tripped",
list="trigger", device=323,
service="urn:micasaverde-com:serviceId:SecuritySensor1", invert=false,
devicename="Bedroom Motion" } },314) <0x778ab520>
50      12/18/18 10:14:49.337   luup_log:313: DelayLight(debug):::
isDeviceOn(323,{ valueOn="1", watched=true, variable="Tripped",
list="trigger", device=323,
service="urn:micasaverde-com:serviceId:SecuritySensor1", invert=false,
devicename="Bedroom Motion" },nil,314) <0x778ab520>
50      12/18/18 10:14:49.338   luup_log:313: DelayLight(debug):::
isDeviceOn() testing 323 val "0" against { 1="1" } (invert=false)
<0x778ab520>
50      12/18/18 10:14:49.339   luup_log:313: DelayLight(debug):::
isDeviceOn() dev 323 checking "0" "=" "1" <0x778ab520>
50      12/18/18 10:14:49.340   luup_log:313: DelayLight(debug):::
isMapDeviceOn({ 85={ valueOn="1", watched=true, variable="Status",
list="on", device=85, service="urn:upnp-org:serviceId:SwitchPower1",
invert=false, devicename="Bedroom Lamp" } },314) <0x778ab520>
50      12/18/18 10:14:49.341   luup_log:313: DelayLight(debug):::
isDeviceOn(85,{ valueOn="1", watched=true, variable="Status", list="on",
device=85, service="urn:upnp-org:serviceId:SwitchPower1", invert=false,
devicename="Bedroom Lamp" },nil,314) <0x778ab520>
50      12/18/18 10:14:49.341   luup_log:313: DelayLight(debug):::
isDeviceOn() testing 85 val "0" against { 1="1" } (invert=false)
<0x778ab520>
50      12/18/18 10:14:49.342   luup_log:313: DelayLight(debug):::
isDeviceOn() dev 85 checking "0" "=" "1" <0x778ab520>
50      12/18/18 10:14:49.344   luup_log:313: DelayLight(debug):::
isMapDeviceOn({ 21={ valueOn="1", watched=true, variable="Status",
list="off", device=21, service="urn:upnp-org:serviceId:SwitchPower1",
invert=false, devicename="Bedroom Light" }, 85={ valueOn="1", watched=true,
variable="Status", list="off", device=85,
service="urn:upnp-org:serviceId:SwitchPower1", invert=false,
devicename="Bedroom Lamp" }, 22={ valueOn="1", watched=true,
variable="Status", list="off", device=22,
service="urn:upnp-org:serviceId:SwitchPower1", invert=false,
devicename="Bedroom Fan" } },314) <0x778ab520>
50      12/18/18 10:14:49.355   luup_log:313: DelayLight(debug):::
isDeviceOn(21,{ valueOn="1", watched=true, variable="Status", list="off",
device=21, service="urn:upnp-org:serviceId:SwitchPower1", invert=false,
devicename="Bedroom Light" },nil,314) <0x778ab520>
50      12/18/18 10:14:49.355   luup_log:313: DelayLight(debug):::
isDeviceOn() testing 21 val "0" against { 1="1" } (invert=false)
<0x778ab520>
50      12/18/18 10:14:49.356   luup_log:313: DelayLight(debug):::
isDeviceOn() dev 21 checking "0" "=" "1" <0x778ab520>
50      12/18/18 10:14:49.357   luup_log:313: DelayLight(debug):::
isDeviceOn(85,{ valueOn="1", watched=true, variable="Status", list="off",
device=85, service="urn:upnp-org:serviceId:SwitchPower1", invert=false,
devicename="Bedroom Lamp" },nil,314) <0x778ab520>
50      12/18/18 10:14:49.358   luup_log:313: DelayLight(debug):::
isDeviceOn() testing 85 val "0" against { 1="1" } (invert=false)
<0x778ab520>
50      12/18/18 10:14:49.359   luup_log:313: DelayLight(debug):::
isDeviceOn() dev 85 checking "0" "=" "1" <0x778ab520>
50      12/18/18 10:14:49.360   luup_log:313: DelayLight(debug):::
isDeviceOn(22,{ valueOn="1", watched=true, variable="Status", list="off",
device=22, service="urn:upnp-org:serviceId:SwitchPower1", invert=false,
devicename="Bedroom Fan" },nil,314) <0x778ab520>
50      12/18/18 10:14:49.361   luup_log:313: DelayLight(debug):::
isDeviceOn() testing 22 val "0" against { 1="1" } (invert=false)
<0x778ab520>
50      12/18/18 10:14:49.361   luup_log:313: DelayLight(debug):::
isDeviceOn() dev 22 checking "0" "=" "1" <0x778ab520>
50      12/18/18 10:14:49.362   luup_log:313: DelayLight(debug):::
startTimer() quiet startup <0x778ab520>

...

50      12/18/18 10:14:59.101   luup_log:313: DelayLight(debug):::
tick("1") luup.device=313 <0x73eab520>
50      12/18/18 10:14:59.101   luup_log:313: DelayLight(debug):::
timerTick(314) <0x73eab520>
50      12/18/18 10:14:59.103   luup_log:313: DelayLight(debug):::
timerTick(314) Status "idle" OffTime 0 <0x73eab520>
50      12/18/18 10:14:59.103   luup_log:313: DelayLight(debug):::
timerTick() timer 314 ("DelayLight Timer 1") nothing more to do, not
rescheduling. <0x73eab520>
50      12/18/18 10:14:59.104   luup_log:313: DelayLight(debug)::: tick()
successful return from timerTick(314) <0x73eab520>
50      12/18/18 10:14:59.109   luup_log:313: DelayLight(debug):::
tick(313) scheduling next master tick for 1545149759(12/18/18.10:15:59)
delay 60 <0x73eab520>

**Comment by toggledbits** [comment]
That's awesome. Thanks for your help and thorough posts/descriptions. Really helps. I'll schedule a release for this weekend (Vera approves plugins on their Monday mornings in Romania).

**Comment by toggledbits** [comment]
I've incorporated the final fix into the stable branch, as well as some other work and code cleanup. If you have the time and inclination, please update to the latest stable branch (all files) and give it a go.

**Comment by brianewman** [comment]
Everything looks great with the latest version.  Thanks again.

**Comment by toggledbits** [comment]
Fix released in 1.8
